res/controllers/common-hid-packet-parser.js

```diff
@@ -415,17 +414,19 @@ class HIDPacket {
             return;
         }
 
-        for (let byte_index = 0; byte_index < bytes; byte_index++) {
-            const index = field.offset + byte_index;
-            if (signed) {
-                if (value >= 0) {
-                    data[index] = (value >> (byte_index * 8)) & 255;
-                } else {
-                    data[index] = 255 - ((-(value + 1) >> (byte_index * 8)) & 255);
-                }
-            } else {
-                data[index] = (value >> (byte_index * 8)) & 255;
-            }
+        const dataView = new DataView(data.buffer);
+        if (field.pack === "b") {
```

Comment:
Where are these values defined? Can we use constants instead of name them literally?

Reply:
Would be character compare faster than a string compare? 
Like 
`
field.pack[0] === 'b'
`

Reply:
I tried this, but it's slower than string compare - can't explain why

Reply:
I'm guessing because indexing into a UTF-8 string is quite expensive? Anyways, goes to show that microoptimization in JS is mostly pointless. Algorithms reign supreme. Caching is great.